### PR TITLE
Add ownerReference to the ShootState resource on creation

### PR DESCRIPTION
**Pull request by swilen-iwanow**
**What this PR does / why we need it**:
Prevents ShootState resources to leak on Shoots deletion.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
`ShootState` resources do now has owner references to their respective `Shoot` resources.
```

### CODE REVIEW

**Comment by plkokanov** [comment]
Looks good, could you just add a release note?

File: pkg/gardenlet/controller/federatedseed/shootstate/common_sync.go [hunk 1]
```diff
@@ -94,8 +95,20 @@ func (s *SyncController) createShootStateSyncReconcileFunc(ctx context.Context,
 		state := extensionObject.GetExtensionStatus().GetState()
 		projectNamespace, shootName := fromNamespace(req.Namespace)
 
+		shoot, err := s.k8sGardenClient.GardenCore().CoreV1beta1().Shoots(projectNamespace).Get(shootName, metav1.GetOptions{})
+		if err != nil {
+			s.log.Error("Couldn't retrieve Shoot %s from namespace %s. Extension %s state with name %s and purpose %v was NOT synced: %v", shootName, projectNamespace, kind, name, purpose, err)
```
Comment (plkokanov) [inline pkg/gardenlet/controller/federatedseed/shootstate/common_sync.go]:
```suggestion
			s.log.Errorf("Couldn't retrieve Shoot %s from namespace %s. Extension %s state with name %s and purpose %v was NOT synced: %v", shootName, projectNamespace, kind, name, purpose, err)
```

**Review by plkokanov — CHANGES_REQUESTED** [review]

**Comment by swilen-iwanow** [comment]
@rfranzke sorry for the delay, fixing my local setup took some effort 😌

**Review by rfranzke — APPROVED** [review]

**Pull request merged**